dbms/src/Client/PerformanceTest.cpp

```diff
@@ -0,0 +1,1556 @@
+                if (sysinfo(system_information))
+                {
+                    std::cerr << "Failed to check system RAM size" << std::endl;
+                    delete system_information;
+                }
+                else
+                {
+                    size_t ram_size_needed = config->getUInt64("preconditions.ram_size");
+                    size_t actual_ram = system_information->totalram / 1024 / 1024;
+                    if (ram_size_needed > actual_ram)
+                    {
+                        std::cerr << "Not enough RAM" << std::endl;
+                        delete system_information;
+                        return false;
+                    }
+                }
+            }
+
+            if (precondition == "table_exists")
+            {
+                String precondition_key = "preconditions.table_exists[" + std::to_string(table_precondition_index++) + "]";
+                String table_to_check = config->getString(precondition_key);
+                String query = "EXISTS TABLE " + table_to_check + ";";
+
+                size_t exist = 0;
+
+                connection.sendQuery(query, "", QueryProcessingStage::Complete, &settings, nullptr, false);
+
+                while (true)
+                {
+                    Connection::Packet packet = connection.receivePacket();
+
+                    if (packet.type == Protocol::Server::Data) {
+                        for (const ColumnWithTypeAndName & column : packet.block.getColumns())
+                        {
+                            if (column.name == "result" && column.column->getDataAt(0).data != nullptr) {
+                                exist = column.column->get64(0);
+                            }
+                        }
+                    }
+
+                    if (packet.type == Protocol::Server::Exception || packet.type == Protocol::Server::EndOfStream)
+                        break;
+                }
+
+                if (exist == 0) {
+                    std::cerr << "Table " + table_to_check + " doesn't exist" << std::endl;
+                    return false;
+                }
+            }
+        }
+
+        return true;
+    }
+
+    void processTestsConfigurations(const Paths & input_files)
+    {
+        tests_configurations.resize(input_files.size());
+
+        for (size_t i = 0; i != input_files.size(); ++i)
+        {
+            const String path = input_files[i];
+            tests_configurations[i] = Config(new XMLConfiguration(path));
+        }
+
+        filterConfigurations();
+
+        if (tests_configurations.size())
+        {
+            Strings outputs;
+
+            for (auto & test_config : tests_configurations)
+            {
+                if (!checkPreconditions(test_config))
+                {
+                    std::cerr << "Preconditions are not fulfilled for test \"" + test_config->getString("name", "") + "\"";
```

Comment:
What we will see in output in this case? Will it contain a note signalling about failed prerequisites?